core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -640,8 +640,8 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
       reconfigurable.validateReconfiguration(newConfigs)
     } catch {
       case e: ConfigException => throw e
-      case _: Exception =>
-        throw new ConfigException(s"Validation of dynamic config update of $updatedConfigNames failed with class ${reconfigurable.getClass}")
+      case e: Exception =>
+        throw new ConfigException(s"Validation of dynamic config update of $updatedConfigNames failed with class ${reconfigurable.getClass} due to: $e")
```

Comment:
Some config exceptions may contain sensitive data like SASL passwords or SSL info. We need to make sure the exception string doesn't include any sensitive information, especially if it could be sent to clients.

Reply:
Thanks for pointing this out, perhaps I'll make this more specific to the error I'm trying to catch

Reply:
This function already catches and re-throws any ConfigExceptions - so it seems a bit unlikely re-throwing the other exceptions missed will return config-related sensitive data. 
However, to be safe 
1. we could limit blast radius by additionally catching IllegalStateException. I've filtered through all the impls of `validateReconfiguration` quickly and this looks to be safe to do.
2. change impls of `validateReconfiguration(Map<String, ?> configs)` to throw ConfigException where they might throw other exception types. e.g. SaslChannelBuilder currently will throw `IllegalStateException` when the SslFactory has not been configured yet - we could wrap this in a ConfigException.

Reply:
Putting a password string into an exception is certainly a bug. If we're doing that, let's file a JIRA for it and fix ASAP.

@rajinisivaram : Do you have an example of code doing this?

There are a limited number of reconfigurables that deal with SASL / SSL so we should be able to look at them all and fix them if needed.

Reply:
We have fixed all known issues with leaked credentials. But we should still be careful about including exception strings related to configuration, where the exception could be from libraries and we are not sure of what they contain, until we have verified that callers have already sanitized these.

Reply:
@ahuang98 Yes, it would be safer to change implementations of validateReconfiguration to throw ConfigException so that unexpected exceptions here can be handled separately. Thanks.